donjo-app/models/Keuangan_grafik_manual_model.php

```diff
@@ -0,0 +1,812 @@
+		}
+		return $data;
+  }
+
+  private function get_subval_belanja( $akun, $thn=false)
+  {
+		$this->db->select('Kelompok, Nama_Kelompok');
+		$this->db->where('Akun', $akun);
+
+		$data = $this->db->get('keuangan_manual_ref_rek2')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_subval_belanja($d['Kelompok'], $thn);
+		  $data[$i]['realisasi'] = $this->real_subval_belanja($d['Kelompok'], $thn);
+		  $data[$i]['sub_belanja2'] = $this->sub_belanja2( $d['Kelompok'], $thn);
+		}
+		return $data;
+  }
+
+  private function get_subval_belanja_bidang( $akun, $thn=false)
+  {
+		$this->db->select('ID_Keg, Nama_Kegiatan');
+    $this->db->like('LEFT(ID_Keg, 10)', $akun, 'after');
+
+    $this->db->order_by('ID_Keg');
+		$data = $this->db->get('keuangan_manual_ref_kegiatan')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_subval_belanja_bidang($d['ID_Keg'], $thn);
+		  $data[$i]['realisasi'] = $this->real_subval_belanja_bidang($d['ID_Keg'], $thn);
+		}
+		return $data;
+  }
+
+  private function get_subval_pembiayaan( $akun, $thn=false)
+  {
+		$this->db->select('Kelompok, Nama_Kelompok');
+		$this->db->where('Akun', $akun);
+
+		$this->db->where('Kelompok', '6.1.');
+		$data = $this->db->get('keuangan_manual_ref_rek2')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_subval_pembiayaan($d['Kelompok'], $thn);
+		  $data[$i]['realisasi'] = $this->real_subval_pembiayaan($d['Kelompok'], $thn);
+		  $data[$i]['sub_pembiayaan2'] = $this->sub_pembiayaan2( $d['Kelompok'], $thn);
+		}
+		return $data;
+  }
+
+  private function get_subval_pembiayaan_keluar( $akun, $thn=false)
+  {
+		$this->db->select('Kelompok, Nama_Kelompok');
+		$this->db->where('Akun', $akun);
+
+		$this->db->where('Kelompok', '6.2.');
+		$data = $this->db->get('keuangan_manual_ref_rek2')->result_array();
+		foreach ($data as $i => $d)
+		{
+		  $data[$i]['anggaran'] = $this->pagu_subval_pembiayaan_keluar($d['Kelompok'], $thn);
+		  $data[$i]['realisasi'] = $this->real_subval_pembiayaan_keluar($d['Kelompok'], $thn);
+		  $data[$i]['sub_pembiayaan_keluar2'] = $this->sub_pembiayaan_keluar2( $d['Kelompok'], $thn);
+		}
+		return $data;
+  }
+
+  private function pagu_subval_pendapatan($kelompok, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 4) AS Kelompok, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $kelompok, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Kelompok');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
+
+  private function pagu_subval_belanja($kelompok, $thn)
+  {
+		$this->db->select('LEFT(Kd_Rincian, 4) AS Kelompok, SUM(Nilai_Anggaran) AS pagu');
+		$this->db->like('Kd_Rincian', $kelompok, 'after');
+		$this->db->where('Tahun', $thn);
+		$this->db->group_by('Kelompok');
+		return $this->db->get('keuangan_manual_rinci')->result_array();
+  }
```

Comment:
Apa beda kedua ini? Mohon periksa semua yg lain -- sepertinya banyak yg duplikat?

Reply:
Kedua fungsi itu berbeda karena pada fungsi akhir akan berbeda jenis akun yang dipakai untuk nilai Pagu/Anggaran nya

Pendapatan
pagu_subval_pendapatan -> get_subval_pendapatan -> -> lap_rp_apbd : $data['pendapatan'] = $this->db->where("Akun = '4.'");

Belanja
pagu_subval_belanja -> get_subval_belanja -> lap_rp_apbd : $data['belanja'] = $this->db->where("Akun = '5.'");